pkg/core/resources/apis/meshmultizoneservice/status_updater_test.go

```diff
@@ -0,0 +1,95 @@
+package meshmultizoneservice_test
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/meshmultizoneservice"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/meshmultizoneservice/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	core_metrics "github.com/kumahq/kuma/pkg/metrics"
+	"github.com/kumahq/kuma/pkg/plugins/resources/memory"
+	test_metrics "github.com/kumahq/kuma/pkg/test/metrics"
+	"github.com/kumahq/kuma/pkg/test/resources/samples"
+)
+
+var _ = Describe("Updater", func() {
+	var stopCh chan struct{}
+	var resManager manager.ResourceManager
+	var metrics core_metrics.Metrics
+
+	BeforeEach(func() {
+		m, err := core_metrics.NewMetrics("")
+		Expect(err).ToNot(HaveOccurred())
+		metrics = m
+		resManager = manager.NewResourceManager(memory.NewStore())
+
+		updater, err := meshmultizoneservice.NewStatusUpdater(logr.Discard(), resManager, resManager, 50*time.Millisecond, m)
+		Expect(err).ToNot(HaveOccurred())
+		stopCh = make(chan struct{})
+		go func() {
+			defer GinkgoRecover()
+			Expect(updater.Start(stopCh)).To(Succeed())
+		}()
+
+		Expect(samples.MeshDefaultBuilder().Create(resManager)).To(Succeed())
+	})
+
+	AfterEach(func() {
+		close(stopCh)
+	})
+
+	It("should add mesh services and port to the status of multizone service", func() {
```

Comment:
Should we also add test for updating mesh service or ports? There is some logic with sorts etc, and we are not checking any of this by starting from scratch and adding service